lib/src/main/scala/spinal/lib/sim/protocolSim/master/Transfer.scala

```diff
@@ -0,0 +1,151 @@
+package spinal.lib.sim.protocolSim.master
+
+import scala.language.implicitConversions
+
+object Transfer {
+
+  /** Get the first element from a sequence of chained transfers
+    *
+    * Transfers are used on the first element to boot or chain. So the first
+    * transfer is returned.
+    *
+    * Transfers of s should be chained by increasing index.
+    *
+    * @see
+    *   chainSeq
+    *
+    * @param ts
+    *   chained transfer sequence
+    * @return
+    *   first transfer of the sequence
+    */
+  implicit def seq2transfer[T <: Transfer](ts: Seq[T]): T = {
+    for (i <- 0 until ts.length - 1)
+      require(ts(i).nextTransfer == Some(ts(i + 1)))
+    ts(0)
+  }
+
+  /** Chains transfers by increasing index
+    *
+    * Transfers of ts must be independent.
+    *
+    * @param ts
+    *   sequence of independent transfers
+    */
+  def chainSeq[T <: Transfer](ts: Seq[T]): Unit = {
+    ts.foreach(t => require(t.nextTransfer.isEmpty))
+    for (i <- 1 until ts.length)
+      ts(0).append(ts(i))
+  }
+}
+
+/** Management of a transfer by a master
+  *
+  * Allows to define and chain transfers into sequences.
+  *
+  *   - The sequence is equivalent to its first transfer.
+  *   - A transfer is equivalent to the sequence starting from it.
+  *
+  * A transfer should not be used twice: for two equivalent transfers to happen,
+  * build the transfer twice. Transfers have internal states.
+  */
+trait Transfer {
+
+  /** Use callback API to run sequence in an async way
+    *
+    * Boots current transfer, which will boot the subsequent transfer of the
+    * sequence, etc.
+    */
+  def boot(): Unit
```

Comment:
:+1: Wilco

Reply:
Well actually, to me boot conveys the idea that I start this one and then it will start the next one etc. Start is just "I start this one". It is related to your other question about transfer vs. sequence. (All is related lol) What do you think about it?

Reply:
>   to me boot conveys the idea that I start this one and then it will start the next one etc. 

Ahhh, this is usage dependent, is see.

so, in some ways, i'm not sure we should bake in the linked list to the Task, but more decouplate things, where a sequance would be a Task which would run multiple sub task sequancialy.

Maybe that the first question to be answer ^^